crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -132,6 +132,87 @@ impl TimelineItemContent {
+    ) -> Self {
+        Self::Message(Message::from_event(c, relations, timeline_items))
+    }
+
+    pub(crate) fn unable_to_decrypt(content: RoomEncryptedEventContent) -> Self {
+        TimelineItemContent::UnableToDecrypt(content.into())
+    }
+
+    pub(crate) fn room_member(
+        user_id: OwnedUserId,
+        full_content: FullStateEventContent<RoomMemberEventContent>,
+        sender: OwnedUserId,
+    ) -> Self {
+        use ruma::events::room::member::MembershipChange as MChange;
+        match &full_content {
+            FullStateEventContent::Original { content, prev_content } => {
+                let membership_change = content.membership_change(
+                    prev_content.as_ref().map(|c| c.details()),
+                    &sender,
+                    &user_id,
+                );
+
+                if let MChange::ProfileChanged { displayname_change, avatar_url_change } =
+                    membership_change
+                {
+                    TimelineItemContent::ProfileChange(MemberProfileChange {
+                        user_id,
+                        displayname_change: displayname_change.map(|c| Change {
+                            new: c.new.map(ToOwned::to_owned),
+                            old: c.old.map(ToOwned::to_owned),
+                        }),
+                        avatar_url_change: avatar_url_change.map(|c| Change {
+                            new: c.new.map(ToOwned::to_owned),
+                            old: c.old.map(ToOwned::to_owned),
+                        }),
+                    })
+                } else {
+                    let change = match membership_change {
+                        MChange::None => MembershipChange::None,
+                        MChange::Error => MembershipChange::Error,
+                        MChange::Joined => MembershipChange::Joined,
+                        MChange::Left => MembershipChange::Left,
+                        MChange::Banned => MembershipChange::Banned,
+                        MChange::Unbanned => MembershipChange::Unbanned,
+                        MChange::Kicked => MembershipChange::Kicked,
+                        MChange::Invited => MembershipChange::Invited,
+                        MChange::KickedAndBanned => MembershipChange::KickedAndBanned,
+                        MChange::InvitationAccepted => MembershipChange::InvitationAccepted,
+                        MChange::InvitationRejected => MembershipChange::InvitationRejected,
+                        MChange::InvitationRevoked => MembershipChange::InvitationRevoked,
+                        MChange::Knocked => MembershipChange::Knocked,
+                        MChange::KnockAccepted => MembershipChange::KnockAccepted,
+                        MChange::KnockRetracted => MembershipChange::KnockRetracted,
+                        MChange::KnockDenied => MembershipChange::KnockDenied,
+                        MChange::ProfileChanged { .. } => unreachable!(),
```

Comment:
Hmm, why would we use the if/let up above and an unreachable here? I know that this has been copied over, but I'll forget to ask about this otherwise.

Reply:
Because this is in the else branch of `if let MChange::ProfileChanged { .. } = membership_change`.

Reply:
Well yes, that's what I'm asking, why even have an else branch instead of just having the match and replacing the unreachable with the block in the if.

Reply:
Because the code is shorter this way, with the `TimelineItemContent::MembershipChange` construction being the same for all the other variants.